src/fileioc/fileioc.asm

```diff
@@ -1379,21 +1362,60 @@ ti_RclVar:
 
 ;-------------------------------------------------------------------------------
 ti_ArchiveHasRoom:
```

Comment:
If we're leaving the input as 24 bits, we should _probably_ do a range check (which effectively checks 16-bit carry for the caller). Also, .entry appears to be unused, so we're not constrained on implementation. I suggest:
```
ti_ArchiveHasRoom:
	pop	de
	pop	bc
	push	bc
	push	de
	cp	a, a	; Set Z flag
	ld	hl, $FF0000
	add	hl, bc
	call	nc, ti.FindFreeArcSpot
	ld	a, 1
	ret	nz
	xor	a, a
	ret
```